### PR TITLE
List alignment

### PR DESCRIPTION
**Pull request by smiska**

### CODE REVIEW

File: src/bulletprooflist.js [hunk 1]
```diff
@@ -4,51 +4,75 @@ var cheerio = require("cheerio");
...
-		var liProcessor = createProcessLiFn(numbered);
-
-		return function processListElem(act) {
-			act.children("li").each(liProcessor);
-
-			var table = $("<table cellpadding=\"0\" cellspacing=\"0\" border=\"0\"></table>");
-
+		})();
+        
+        var liProcessor = (function (numbered) {
```
Comment (gyulanemeth) [inline src/bulletprooflist.js]: you introduced an unnecessary scope here.

File: src/bulletprooflist.js [hunk 2]
```diff
@@ -4,51 +4,75 @@ var cheerio = require("cheerio");
...
+            var bullet = ((function listElemMarker() {
```
Comment (gyulanemeth) [inline src/bulletprooflist.js]: I also don't understand why you need this scope here.

File: src/bulletprooflist.js [hunk 3]
```diff
@@ -4,51 +4,75 @@ var cheerio = require("cheerio");
...
+            const alignment = alignmentProp ? "align=\"" + alignmentProp + "\"" : "";
+
+            var table = $("<table cellpadding=\"0\" cellspacing=\"0\" border=\"0\" " + alignment + "></table>");
+
+            // set back alignments for the next list
+            tableAligns = [];
+            
 			table.html(act.html());
-			act.replaceWith(table);
+            act.replaceWith(table);
```
Comment (gyulanemeth) [inline src/bulletprooflist.js]: tabs vs spaces.

File: src/bulletprooflist.js [hunk 4]
```diff
@@ -4,51 +4,75 @@ var cheerio = require("cheerio");
...
-				return "&#x02022;"; //not good: Lotus Notes 7, Outlook 2013
-				//return "&#8226;"; //not good: Lotus Notes 7, Outlook 2013
-				//return "&bull;";
-				//return "&bullet;";
+    
+
+	function listElemProcessorFn(numbered) {
+        var tableAligns = [];
+        
+        var alignmentModifier = (function () {
```
Comment (gyulanemeth) [inline src/bulletprooflist.js]: why do you need to wrap this into a function which is executed immediately???

File: src/bulletprooflist.js [hunk 5]
```diff
@@ -4,51 +4,75 @@ var cheerio = require("cheerio");
...
+            };
+        })(numbered);
+
+		return function processListElems(act) {
+            
+            act.children("li").each(alignmentModifier);
+            act.children("li").each(liProcessor);
+
+            // CHECK LIST ITEM ALIGNMENT
+            var equalityChecker = tableAligns.every((v, i, a) => i === 0 || v === a[i - 1] );
```
Comment (gyulanemeth) [inline src/bulletprooflist.js]: by the name of the variable, I would guess it's a function. Also, you use var and const... why don't you use const all the time?

**Review by gyulanemeth — CHANGES_REQUESTED** [review]

File: src/bulletprooflist.js [hunk 6]
```diff
@@ -4,51 +4,66 @@ var cheerio = require("cheerio");
...
-		return function processLi(idx, actLiElem) {
-			var act = $(actLiElem);
+		return function processedList(act) {
+			act.children("li").each(processListElem);
 
-			var actContent = act.html();
+			// CHECK LIST ITEM ALIGNMENT
+			var equalityChecker = tableAligns.every((v, i, a) => i === 0 || v === a[i - 1] );
+			var alignmentProp = equalityChecker && tableAligns.length !== 0 ? tableAligns[0] : "";
+			var alignment = alignmentProp ? "align=\"" + alignmentProp + "\"" : "";
```
Comment (gyulanemeth) [inline src/bulletprooflist.js]: you can add a space here, so you can remove the unnecessary one later.

File: src/bulletprooflist.js [hunk 7]
```diff
@@ -4,51 +4,66 @@ var cheerio = require("cheerio");
...
 
-			var tr = $("<tr></tr>");
+			var table = $("<table cellpadding=\"0\" cellspacing=\"0\" border=\"0\" " + alignment + "></table>");
```
Comment (gyulanemeth) [inline src/bulletprooflist.js]: here.

File: src/bulletprooflist.js [hunk 8]
```diff
@@ -4,51 +4,66 @@ var cheerio = require("cheerio");
...
+		})(numbered);
 
-		return function processLi(idx, actLiElem) {
-			var act = $(actLiElem);
+		return function processedList(act) {
+			act.children("li").each(processListElem);
 
-			var actContent = act.html();
+			// CHECK LIST ITEM ALIGNMENT
+			var equalityChecker = tableAligns.every((v, i, a) => i === 0 || v === a[i - 1] );
```
Comment (gyulanemeth) [inline src/bulletprooflist.js]: what about v === a[0] in the every? Also, why did you use v, i and a?

**Review by gyulanemeth — CHANGES_REQUESTED** [review]

**Comment by smiska** [comment]
@gyulanemeth 
please allocate time to debug sometime

**Pull request closed without merging**